internal/pkg/template/templates/workloads/jobs/scheduled-job/manifest.yml

```diff
@@ -34,8 +34,8 @@ image:
 
 cpu: {{.CPU}}       # Number of CPU units for the task.
 memory: {{.Memory}}    # Amount of memory in MiB used by the task.
-{{- if .Platform}}
-platform: {{.Platform}}   # See https://aws.github.io/copilot-cli/docs/manifest/scheduled-job/#platform
+{{- if .Platform.PlatformString}}
```

Comment:
There is a hidden assumption here that the `.Platform` field is always non-nil. This is true because of the `cli` pkg but it's not obvious. These templates are rendered by the `manifest` pkg so it should be its responsibility to ensure that `Platform` always has a value.  


Can we maybe add this behavior in the default constructors?
```go
newDefaultBackendService() {
   ...
   BackendServiceConfig: BackendServiceConfig {
       ...,
       Platform: &PlatformArgsOrString{}, // Ensure .Platform is always set.
   }
}
```

Reply:
Why don't we set it to be struct instead of pointer of struct?

Reply:
It's small, but for space considerations? We have other pointers to structs for other default values, too.

Reply:
It's not small now lol. we are now trying to make it consistent to always use non pointer unless it is necessary. For this one using non pointer we can be sure that .Platform is always set.